txcache/transactionsHeapItem.go

```diff
@@ -123,16 +126,17 @@ func (item *transactionsHeapItem) detectWillFeeExceedBalance() bool {
 		return false
 	}
 
-	futureAccumulatedFee := new(big.Int).Add(item.accumulatedFee, fee)
+	// Here, we are not interested into an eventual transfer of value (we only check if there's enough balance to pay the transaction fee).
+	futureConsumedBalance := new(big.Int).Add(item.consumedBalance, fee)
 	senderBalance := item.senderState.Balance
 
-	willFeeExceedBalance := futureAccumulatedFee.Cmp(senderBalance) > 0
+	willFeeExceedBalance := futureConsumedBalance.Cmp(senderBalance) > 0
```

Comment:
We don't detect invalid transactions. We always assume that the value to be transferred is actually transferred (even if it isn't, actually, during processing).

We'll stop selecting transactions from a given sender once we detect (under the assumption above) that she doesn't have enough balance to pay the fee for the "currentTransaction".